collections/_article/progress-report-web-export-in-4-3.md

```diff
@@ -0,0 +1,311 @@
+
+### Betting on the wrong horse
+[`SharedArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)s
+were supposed to revolutionize the Web. And they did. That API permits to share memory between
+[Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers)
+(which are the Web native "threads").
+
+Unfortunately, we live in the same timeline that of [Spectre and Meltdown exploits](https://meltdownattack.com/).
+
+The result of this is that browsers greatly reduced where and how you can use that API. Namely, browsers nowadays require
+websites to be [cross-origin isolated](https://web.dev/coop-coep/). When isolated, it does unlock the potential of
+`SharedArrayBuffer`s, but at what cost? The capacity to make remote calls to other websites.
+
+_Adieu_ game monetization. _Adieu_ interacting with an external API.
+
+!["What did it cost? Everything." exchange between Gamora and Thanos in Avengers Infinity](/storage/blog/progress-report-web-export-in-4-3/thanos-gamora.gif)
+
+The thing is that during the development of Godot 4, misconceptions about the Web platform made it so that we bet on
+the wrong horse: we went all-in with using threads with `SharedArrayBuffer`s. We minimized the cost of using
+cross-origin isolated websites. Sadly, it made it so that
+[Winterpixel Games](https://www.winterpixel.com/) (makers of [Rocket Bot Royale](https://rocketbotroyale.winterpixel.io/)
+and [Goober Dash](https://gooberdash.winterpixel.io/)) were unfortunately stuck on Godot 3.x, which supports
+running games on the main thread.
+
+### Backporting the single-threaded feature from Godot 3.x
+At the end of the development of Godot 4.2, [the Foundation](https://godot.foundation/) tasked me to find a way to backport
+the ability to build Godot without using threads in order to export Godot games without the pesky
+<abbr title="Cross-Origin-Opener-Policy">COOP</abbr>/<abbr title="Cross-Origin-Embedder-Policy">COEP</abbr> headers.
+
+This was not easy because 4.0 was made with threads in mind. It maybe explains why we were so hesitant to go ahead with
+this change. Fortunately, with the help of our thread guru
+[Pedro J. Estébanez (@RandomShaper)](https://github.com/RandomShaper), I was able to tame the beast and put back single-
+threaded builds.
+
+[My <abbr title="Pull Request">PR</abbr>](https://github.com/godotengine/godot/pull/85939) was finally merged at the
+beginning of the Godot 4.3 development period.
+
+![Web export thread support option](/storage/blog/progress-report-web-export-in-4-3/thread-support-export-option.webp)
+
+It even had some unexpected benefits. Apple devices (macOS and iOS) were long known to have issues when playing Godot
+Web exports. Well, when you do export your game single-threaded, these issues fortunately disappear.
+
+But we couldn't really celebrate that much, as it introduced a new bug. A bug virtually unsolvable…
+
+…until now.
+
```

Comment:
Explain the actual bug here pls, or move the next headline lower down

Reply:
I moved the headline lower down.